src/js/stores/FiltersStore.js

```diff
@@ -66,6 +49,14 @@ var FiltersStore = Reflux.createStore({
         this.trigger();
     },
 
+    removeUnavailableFilters() {
+        _.forEach(currentFilters, (f) => {
```

Comment:
You should user `Array.filter()` and then trigger(), because with the call to `delete()` you trigger on each matching iteration